struct.go

```diff
@@ -329,15 +329,40 @@ func (k K) String() string {
 	}
 }
 
-// Message is malformated or invalid
-var ErrBadMsg = errors.New("Bad Message")
-
 // Message header is invalid
 var ErrBadHeader = errors.New("Bad header")
 
 // Cannot process sync requests
 var ErrSyncRequest = errors.New("nosyncrequest")
 
+// NewQError initializes a new QError with a message.
+func NewUnsupportedTypeError(msg string) error {
+	return &UnsupportedTypeError{msg}
+}
+
+// UnsupportedTypeError is an error returned by Fn() calls. It can be used to distinguish these errors from generic errors
+type UnsupportedTypeError struct {
+	msg string
+}
+
+func (e *UnsupportedTypeError) Error() string {
+	return e.msg
+}
+
+// NewQError initializes a new QError with a message.
+func NewQError(msg string) error {
+	return &QError{msg}
+}
+
+// QError is an error returned by Fn() calls. It can be used to distinguish these errors from generic errors
+type QError struct {
```

Comment:
Could we rename this to something that doesn't have `Q` in it? My reasoning is that q is the language and not the actual runner. So in a sense, there is no `GoError` or `JavaError`. I think another alternative is `Error`(`kdb.Error`), or `RuntimeError`(`kdb.RuntimeError`)